package/Codegen/Program.cs

```diff
@@ -367,6 +367,15 @@ private void DumpTypes(Version version)
             var propertiesGen = new TypeProperties(properties, fakeProps, syntheticProps).TransformText();
             var enumsGen = new TypeEnums().TransformText();
 
+            // order enum list by namespace
+            MrType[] list = new MrType[Util.enumsToGenerateConvertersFor.Count];
+            Util.enumsToGenerateConvertersFor.CopyTo(list);
+            var orderedList = list.OrderBy(item => Util.GetTSNamespace(item));
+            Util.enumsToGenerateConvertersFor.Clear();
+            foreach (var orderedItem in orderedList) {
+                Util.enumsToGenerateConvertersFor.Add(orderedItem);
+            }

```

Comment:
you can simplify this:
```csharp
            Util.enumsToGenerateConvertersFor = Util.enumsToGenerateConvertersFor.OrderBy(e => Util.GetTSNamespace(e));
```

Then change `enumsToGenerateConvertersFor`  to be declared as `IEnumerable<MrType>` and change the `Add`s to `Append`s

Reply:
Thank you for your detailed suggestion.

I tried 
1. changing the declaration to:
`public static IEnumerable<MrType> enumsToGenerateConvertersFor = new HashSet<MrType>();`

2. Changed the code here to:
`Util.enumsToGenerateConvertersFor = Util.enumsToGenerateConvertersFor.OrderBy(e => Util.GetTSNamespace(e));`

3. changed all the Adds to Append (Found one occurrence in Util.cs and one in Program.cs)

But after that the enumsToGenerateConvertersFor is empty. Is it because of the issue discussed [here?](https://stackoverflow.com/questions/1210295/how-can-i-add-an-item-to-a-ienumerablet-collection) Maybe Append won't work?

Reply:
good catch. How's this: https://github.com/microsoft/react-native-xaml/compare/orderEnums?expand=1

Reply:
Hi @asklar Thanks a lot for sharing the improved implementation. I can confirm that it generates the new Enums.ts file as expected and also that it resolves our duplicate declaration issue. I have updated the code in this PR as well based on your feedback.